academic_helper/views/schedule.py

```diff
@@ -30,7 +36,25 @@ def get(self, request, *args, **kwargs):
             # get_context_data, which already used this cookie
             response.delete_cookie(SCHEDULE_COOKIE)
 
-        return response
+        if request.GET.get("download") == "ical":
+            ical = ScheduleExport(user=self.user).as_ical()
+            buffer = BytesIO()
+            buffer.writelines([line.encode() for line in ical])
+            buffer.seek(0)
+            return FileResponse(
+                buffer, as_attachment=True, filename=f"coursist_export_{self.user}_{arrow.utcnow()}.ics",
+            )
+        elif request.GET.get("download") == "json":
+            event_json = ScheduleExport(user=self.user).as_dict()
+            buffer = BytesIO()
+            buffer.write(json.dumps(event_json, indent=4, sort_keys=True, cls=DjangoJSONEncoder).encode())
+            buffer.seek(0)
+            return FileResponse(
+                buffer, as_attachment=True, filename=f"coursist_export_{self.user}_{arrow.utcnow()}.json",
+            )
```

Comment:
Put these before the first lines of the function (no need to generate `response` if it not used, right?).

Reply:
Good point - will fix.

Reply:
Done in https://github.com/asaf-kali/coursist/pull/113/commits/47a24458496f700d47ad8dd3bba70fa2b9ef9cd6